### PR TITLE
Added perPeerRateLimit env config

### LINKED ISSUE #1579: Is there a way to dynamically configure `DefaultPerPeerRateLimit`?
https://github.com/libp2p/go-libp2p/blob/5c297a87002a04a68143c59167502a9be6c73f2f/p2p/net/swarm/limiter.go#L57

### PR DESCRIPTION
**Pull request by millken**
Fix #1579

### CODE REVIEW

**Review by vyzo — APPROVED** [review]

**Comment by marten-seemann** [comment]
Not a big fan of making this configurable via environment variable.
The best fix would be to make the swarm injectable, and then use a constructor option. The next-better fix would be to have a package-level global variable.

**Comment by MarcoPolo** [comment]
> Not a big fan of making this configurable via environment variable.
> The best fix would be to make the swarm injectable, and then use a constructor option. The next-better fix would be to have a package-level global variable.

Agreed. @millken mind making the package level global  change please? It’s better since then this is exposed prominently in the pkg reference, and your code can read your env var and set the variable.

**Comment by millken** [comment]
> > Not a big fan of making this configurable via environment variable.
> > The best fix would be to make the swarm injectable, and then use a constructor option. The next-better fix would be to have a package-level global variable.
> 
> Agreed. @millken mind making the package level global change please? It’s better since then this is exposed prominently in the pkg reference, and your code can read your env var and set the variable.

Updated

File: config/config.go [hunk 1]
```diff
@@ -104,6 +104,9 @@ type Config struct {
 
 	EnableHolePunching  bool
 	HolePunchingOptions []holepunch.Option
+
+	FDLimit      int
+	PerPeerLimit int
```
Comment (marten-seemann) [inline config/config.go:109]: There's no way for a libp2p user to directly set values in this config.

**Review by marten-seemann — CHANGES_REQUESTED** [review]

**Review by marten-seemann — CHANGES_REQUESTED** [review]
I don't think we should have global config options for this. This seems overly specific.
Why don't you just add a variable, as @MarcoPolo suggested in https://github.com/libp2p/go-libp2p/pull/1580#issuecomment-1145483558.

**Comment by millken** [comment]
> I don't think we should have global config options for this. This seems overly specific. Why don't you just add a variable, as @MarcoPolo suggested in [#1580 (comment)](https://github.com/libp2p/go-libp2p/pull/1580#issuecomment-1145483558).

Ok, Just change `const DefaultPerPeerRateLimit = 8` to `var DefaultPerPeerRateLimit = 8`

**Review by MarcoPolo — APPROVED** [review]

**Comment by MarcoPolo** [comment]
Thanks!

**Pull request merged**